File: source/downloads/code/LeetCode/Java/H-index2.java
public class Solution {
    public int hIndex(int[] citations) {
        int r = citations.length-1;
        int l = 0;
        while(l<=r){
            int m = l+(r-l)/2;
            if(citations[m] < citations.length-m)
               l = m+1;
            else
               r = m-1; 
        }
        return citations.length-l;
    }
}

File: downloads/code/LeetCode/Python/H-Index2.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        start = 0
        end = len(citations)-1

        while start <= end:
            mid = (start + end)/2
            if citations[mid] < len(citations)-mid:
                start = mid+1
            else:
                end = mid-1
        return len(citations)-start
